src/shims/unix/fs/epoll.rs

```diff
@@ -0,0 +1,64 @@
+// use crate::shims::unix::fs::FileDescriptor;
+// use crate::shims::unix::fs::SeekFrom;
+use crate::*;
+
+use super::FileDescriptor;
+use super::FileHandle;
+
+use std::collections::HashMap;
+use std::io;
+use std::io::SeekFrom;
+
+#[derive(Clone, Debug, Default)]
+pub struct Epoll {
+    pub file_descriptors: HashMap<i32, ()>,
+}
+
+impl FileDescriptor for Epoll {
+    fn as_file_handle<'tcx>(&self) -> InterpResult<'tcx, &FileHandle> {
+        throw_unsup_format!("epoll file descriptors cannot be used as FileHandle");
+    }
+
+    fn as_epoll_handle<'tcx>(&mut self) -> InterpResult<'tcx, &mut Epoll> {
+        Ok(self)
+    }
+
+    fn read<'tcx>(
+        &mut self,
+        _communicate_allowed: bool,
+        _bytes: &mut [u8],
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        throw_unsup_format!("cannot read from epoll file descriptors");
+    }
+
+    fn write<'tcx>(
+        &self,
+        _communicate_allowed: bool,
+        _bytes: &[u8],
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        throw_unsup_format!("cannot write from epoll file descriptors");
+    }
+
+    fn seek<'tcx>(
+        &mut self,
+        _communicate_allowed: bool,
+        _offset: SeekFrom,
+    ) -> InterpResult<'tcx, io::Result<u64>> {
+        throw_unsup_format!("cannot seek on epoll file descriptors");
+    }
+
+    fn close<'tcx>(
+        self: Box<Self>,
+        _communicate_allowed: bool,
+    ) -> InterpResult<'tcx, io::Result<i32>> {
+        throw_unsup_format!("epoll file descriptors cannot be closed");
```

Comment:
"When no longer required, the file descriptor returned by epoll_create() should
be closed by using [close(2)](https://man7.org/linux/man-pages/man2/close.2.html)."

https://man7.org/linux/man-pages/man2/epoll_create1.2.html

Reply:
yea, this PR is *very* WIP. We want to have a code path that hits this before implementing it